### ISSUE TITLE
Travis: parallelize functional test in sub-jobs

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#3287]
As far as I understand, travis runs sub-job in parallel:

![imagen](https://user-images.githubusercontent.com/1534240/45217063-03533c00-b2a3-11e8-94a8-1053a8000d3a.png)

Could we exploit this in order to paralelize functional testing? I mean, use the `--fromIx` and `--toIx` in order to run diferent blocks in different sub-jobs.

For instance, if we chose 100 tests as block sice we can have:

* functional1 sub-job with test from 1 to 100 (`./testHarnes --fromIx 1 --toIx 100`)
* functional2 sub-job with test from 101 to 200 (`./testHarnes --fromIx 101 --toIx 200`)
...
* functional11 sub-job with test from 1000 to 1100 (`./testHarnes --fromIx 1000 --toIx 1100`). Actually, we currently have 1030 but this will work. 

When we reach teset 1101 a functional12 sub-job will be created.

If this is feasible, it will speed up **a lot** the time travis takes to finish its job.

@caa06d9c what do you think?

**Comment by fgalan** [comment]
I'm realizing that testHarness doesn't have a `--toIx` CLI...

```
fermin@neodeb:~/src/fiware-orion/test/functionalTest$ ./testHarness.sh -u
vie sep  7 13:42:08 CEST 2018
Usage: testHarness.sh [-u (usage)]
                      [-v (verbose)]
                      [--filter <test filter>]
                      [--match <string for test to match>]
                      [--keep (don't remove output files)]
                      [--dryrun (don't execute any tests)]
                      [--dir <directory>]
                      [--fromIx <index of test where to start>]
                      [--ixList <list of test indexes>]
                      [--skipList <list of indexes of test cases to be skipped>]
                      [--stopOnError (stop at first error encountered)]
                      [--no-duration (removes duration mark on successful tests)]
                      [--noCache (force broker to be started with the option --noCache)]
                      [--cache (force broker to be started without the option --noCache)]
                      [--noThreadpool (do not use a threadpool, unless specified by a test case. If not set, a thread pool of 200:20 is used by default in test cases which do not set notificationMode options)]
                      [ <directory or file> ]*
```

However, if the division of jobs described above is possible, it think it should worth the effort to implement `--toIx` in testHarness.sh. It shouldn't be too dificuult.

**Comment by kzangeli** [comment]
--toIx could be implemented in 30 min, testing included

**Comment by fgalan** [comment]
@caa06d9c not sure of understanding what you mean...

So you mean that each travis sub-job would run in parallel in the *same docker container*, i.e. sharing ports, filesystem, etc.? Or, on the contrary, each travis sub-job uses a *different docker container*, i.e. isolated ports, isolate filesytem, etc. for each sub-job.

**Comment by fgalan** [comment]
> --toIx could be implemented in 30 min, testing included

Done in PR https://github.com/telefonicaid/fiware-orion/pull/3457

I'll play with .travis.yml trying to make smaller blocks in the ft job.

**Comment by fgalan** [comment]
I  have had a look to .travis.yml and I afraid is harder than I originally think... I understand this is the part in which the jobs are configured:

```
  matrix:
    - TEST=compliance
    - TEST=unit
    - TEST=functional
```

but I don't know how to provide the --fromIx and --toIx parameters.

@caa06d9c could you have a look and help, pls? Thanks!

**Comment by caa06d9c** [comment]
`/ci/rpm7/build.sh` should be fixed, so new parameters should be added and `FUNCTIONAL TEST` too to pass this params and   `/ci/rpm7/makefile` target too

**Comment by fgalan** [comment]
I think we have hit the limits with PR https://github.com/telefonicaid/fiware-orion/pull/3498. From now on, travis is broken :(

@caa06d9c did you finally solved the way of parallelizing test harness, please?

**Comment by caa06d9c** [comment]
Yes, please, take a look into [Orion-LD repository](https://github.com/FIWARE/context.Orion-LD/tree/develop/docker) and its Travis [build](https://travis-ci.org/FIWARE/context.Orion-LD/builds/535356088). I can do the same for this repository. but you can see, that ci things were refactored.

**Comment by fgalan** [comment]
> I can do the same for this repository. 

Yes, please. Thanks!
